### PR TITLE
Implement automated tests on push and checks before commit

### LINKED ISSUE #8: Automate repo code checking and tests
To make sure all code is working, we should automate our tests to be run before merging.
This will prevent any non-working tests to be present on the main branch.
The tests will report the coverage.

Also there should be pre-commit checks to validate all pep rules.

### PR DESCRIPTION
**Pull request by gillesvink**

### CODE REVIEW

File: tests/test_setup.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""Simple script to test the CICD workflow."""
```
Comment (gillesvink) [inline tests/test_setup.py:1]: This file is needed to pass the Github action

**Review by cglukas — COMMENTED** [review]
Nicely done ✅ You're really into this ci cd game 😁

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          echo "Current directory is ${GITHUB_WORKSPACE}."
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          echo "Installed pip packages."
+
+      - name: Run pytest
+        run: pytest tests --doctest-modules --junitxml=junit/test-results-${{ matrix.python-version }}.xml
```
[the remark this answers is not in the document]
Comment (gillesvink) [reply .github/workflows/tests.yaml]: Is included by default to give as a report that can be exported. However, I changed it now to be html so its more readable

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.3.7
+    hooks:
+      - id: ruff
+        args: [--fix]
```
[the remark this answers is not in the document]
Comment (gillesvink) [reply .pre-commit-config.yaml:6]: Yes! Added it now

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+run-name: Test code
+on:
+  push:
+
+jobs:
+  pytest:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12"]
```
[the remark this answers is not in the document]
Comment (gillesvink) [reply .github/workflows/tests.yaml]: 3.7 is nuke 13, 3.9 is nuke 14, 3.10 is nuke 15 and 3.11 will be nuke 16 (as in vfx-reference). Removed 3.12 and 3.8 now

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+name: Test NukeTesting codebase and upload test results.
+run-name: Test code
+on:
+  push:
+    paths:
+      - '**.py'
```
Comment (cglukas) [inline .github/workflows/tests.yaml:6]: Didn't know this feature 😍

**Review by cglukas — APPROVED** [review]

**Pull request merged**